src/main/scala/is/hail/utils/NumericPair.scala

```diff
@@ -0,0 +1,52 @@
+package is.hail.utils
+
+import scala.math.Numeric.{DoubleIsFractional, FloatIsFractional, IntIsIntegral, LongIsIntegral}
+
+trait NumericPair[T, BoxedT >: Null] extends Serializable {
+
+  def box(t: T): BoxedT
+
+  def unbox(u: BoxedT): T
+
+  implicit val numeric: scala.math.Numeric[T]
+}
+
+object NumericPairs {
```

Comment:
I'd make this a trait and have utils package object inherit from it.  Probably call it NumericPairImplicits or something.